FhirToDataLake/scripts/Set-SynapseEnvironment.ps1

```diff
@@ -218,20 +221,155 @@ function New-TableAndViewsForResources
         Remove-Job -Job $finishedJob
 
         if ($finishedJob.State -eq 'Failed') {
             Write-Host " -> $($finishedJob.ChildJobs[0].JobStateInfo.Reason.Message)" -ForegroundColor Red
-            Get-Job -Name $jobName | Wait-Job | Remove-Job | Out-Null
+            Get-Job -Name $JobName | Wait-Job | Remove-Job | Out-Null
             throw "Creating Table and Views job failed: $($finishedJob.ChildJobs[0].JobStateInfo.Reason.Message)"
         }
     }
 }
 
+
+function Get-CustomizedSchemaImage {
+    param ([string]$orasAppPath, [string]$schemaImageReference, [string]$CustomizedTemplateDirectory)
+
+    $registryName = $schemaImageReference.Substring(0, $schemaImageReference.IndexOf('.azurecr.io'))
+    Connect-AzContainerRegistry -Name $registryName -ErrorAction stop
+
+    # Leverage the oras to pull the image from Container Registry.
+    $orasParameters = @(
+        'pull'
+        $schemaImageReference
+    )
+    $response = & $orasAppPath $orasParameters 2>&1
+    
+    if ($LastExitCode -ne 0)
+    {
+        Write-Host $response -ForegroundColor Red
+        throw "Failed to pull the customized schema image: $response" 
+    }
+
+    $compressPackages = Get-ChildItem -Path * -Include '*.tar.gz' -Name
+    Write-Host "Successfully pull the customized schema image: $compressPackages" -ForegroundColor Green
+
+    # Unpack the image compressed package to customized template directory    
+    if (!(Test-Path $CustomizedTemplateDirectory)){
+        New-Item $CustomizedTemplateDirectory -ItemType Directory
+    }
+
+    foreach ($compressPackage in $compressPackages){
+        $unpackParameters = @(
+            '-zxvf'
+            $compressPackage
+            '-C'
+            $CustomizedTemplateDirectory
+        )
+        $response = & 'tar' $unpackParameters 2>&1
+
+        if ($LastExitCode -ne 0)
+        {
+            Write-Host $response -ForegroundColor Red
+            throw "Failed to unpack customized schema image: $response" 
+        }
+    }
+}
+
+function Get-CustomizedSchemaType {
+    param ([string]$resourceType)
+
+    return "$($resourceType)_Customized"
+}
+
+function Get-CustomizedTableSql {
+    param ([string]$schemaType, [Hashtable]$jsonSchemaObject)
+
+    $customizedTableProperties = ""
+    foreach ($property in $jsonSchemaObject['properties'].GetEnumerator()){
+        $sqlType = switch($property.Value['type']) 
+        {
+            'number' { 'float' ; Break }
+            'integer' { 'bigint' ; Break }
+            'boolean' { 'bit' ; Break }
+            'string' { 'NVARCHAR(4000)' ; Break }
+            Default {
+                Write-Host "Invalid property type in '$($schemaType).$($property.Name)': $($property.Value['type'])"
+                throw
+            }
+        }
+
+        $customizedTableProperties += "    [$($property.Name)] $sqlType,"
+    }
+    
+    $createCustomizedTableSql = "CREATE EXTERNAL TABLE [fhir].[$($schemaType)] (
```

Comment:
just curios why we need additional $ here?

Reply:
Removed duplicated $, thanks.